daemons/controld/controld_join_dc.c

```diff
@@ -651,15 +654,18 @@ finalize_join_for(gpointer key, gpointer value, gpointer user_data)
         return;
     }
 
-    // Acknowledge node's join request
-    crm_debug("Acknowledging join-%d request from %s",
-              current_join_id, join_to);
+    // Acknowledge or nack node's join request
+    crm_debug("%sing join-%d request from %s",
+              integrated? "Acknowledg" : "Nack", current_join_id, join_to);
     acknak = create_dc_message(CRM_OP_JOIN_ACKNAK, join_to);
-    pcmk__xe_set_bool_attr(acknak, CRM_OP_JOIN_ACKNAK, true);
-    crm_update_peer_join(__func__, join_node, crm_join_finalized);
-    pcmk__update_peer_expected(__func__, join_node, CRMD_JOINSTATE_MEMBER);
+    pcmk__xe_set_bool_attr(acknak, CRM_OP_JOIN_ACKNAK, integrated);
 
-    send_cluster_message(crm_get_peer(0, join_to), crm_msg_crmd, acknak, TRUE);
+    if (integrated) {
+        // No change needed for a nacked node
+        crm_update_peer_join(__func__, join_node, crm_join_finalized);
+        pcmk__update_peer_expected(__func__, join_node, CRMD_JOINSTATE_MEMBER);
```

Comment:
I plan to keep this here but the expected state should already be member from `do_dc_join_filter_offer()`.